kele.el

```diff
@@ -1116,10 +1123,142 @@ Only populated if Embark is installed.")
       (kele--disable)
     (kele--enable)))
 
+(defun kele--namespace-read (prompt initial-input history)
+  "Read a namespace value using PROMPT, INITIAL-INPUT, and HISTORY.
+
+Assumes that the current Transient prefix's :scope is an alist w/ `context' key."
+  ;; FIXME: Make this resilient to the prefix's scope not having a context
+  ;; value.  If not present (or the scope is not an alist or the scope is not
+  ;; defined), default to current context.
+  (if-let ((context (alist-get 'context (oref transient--prefix scope))))
+      (completing-read
+       prompt
+       (-cut kele--resources-complete <> <> <>
+             :cands (kele--get-namespaces
+                     (alist-get 'context (oref transient--prefix scope)))
+             :category 'kele-namespace)
+       nil t initial-input history)
+    (error "Unexpected nil context in `%s'" (oref transient--prefix command))))
```

Comment:
`kele--namespace-read` needs to be useable independent of Transient.

Let's have this function inject the context, then the calling Transient infix/prefix/whatever can curry the context value in via `apply-partially` or `-cut` or what have you.

Reply:
Eh, fine to keep as-is. Just rename it to make it obvious that this is specifically to be used w/ a Transient prefix.